### PR TITLE
Add section about integration with CI tools to the Readme.

### LINKED ISSUE #712: Travis integration
Please consider adding TravisCI integration documentation in the readme.
Thank you.

### PR DESCRIPTION
**Pull request by jrfnl**
For now, it only covers running WPCS on Travis without using Composer.

I imagine that in the future sub-sections might be added for Circle CI, Srutinizer etc.

Fixes #712

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
...
+
+#### Travis CI example
+```bash
+language: php
+
+matrix:
+    include:
+        # Arbitrary PHP version to run the sniffs against.
+        - php: '7.0'
+          env: SNIFF=1
```
Comment (szepeviktor) [inline README.md]: These lines are not two-space indented.

Comment (jrfnl) [inline README.md]: Is that a requirement ? I've taken the code from an existing travis file and never had a problem with it the way it is now (as long as the indentation is consistent)

Comment (szepeviktor) [inline README.md]: Later on we have 2 spaces.

Comment (jrfnl) [inline README.md]: Ok, I've fixed it up (redone the commit so as to keep a clean history)

Comment (szepeviktor) [inline README.md]: You may freely commit as many as you want.
GitHub has a squash feature.

Comment (jrfnl) [inline README.md]: True, but it's not always used ;-)

File: README.md [hunk 2]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
 
 Please see “[Setting up PHP CodeSniffer in Visual Studio Code](https://tommcfarlin.com/php-codesniffer-in-visual-studio-code/)”, a tutorial by Tom McFarlin.
 
+
+## Running your code through WPCS automatically using CI tools
+
+### [Travis CI](https://travis-ci.org/)
+
+To integrate PHPCS with WPCS with Travis, you'll need to install both `before_script` (or at `install`) and add the run command to the `script`.
```
Comment (ntwb) [inline README.md]: "... with Travis CI" please, also PHPCS should be added in the `before_install` section of the `.travis.yml`, this way if any errors occur the build job will report `Errored` rather than `Failed` this is important as it determines there was an issue setting up your environment rather than the code you are actually testing against with PHPCS.

So something like:  "To integrate PHPCS with WPCS with Travis CI, you'll need to install both in the `before_install` section of your `.travis.yml` configuration and add the run command to the `script`."

File: README.md [hunk 3]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
...
+```bash
+language: php
+
+matrix:
+  include:
+    # Arbitrary PHP version to run the sniffs against.
+    - php: '7.0'
+      env: SNIFF=1
+          
+before_script:
```
Comment (ntwb) [inline README.md]: Per the above note, changing this to `before_install` will be 👌

File: README.md (hunk 2 above)
Comment (jrfnl) [inline README.md]: Interesting! I didn't know that. I've changed the text & example code.

File: README.md (hunk 1 above)
Comment (ntwb) [inline README.md]: p.s. @jrfnl YAML is a subset JSON spec and need to use spaces not tabs for indentation, I think most parsers will detect tabs if used but spaces will always be safer 😬 

See also https://core.trac.wordpress.org/browser/trunk/.editorconfig#L16

Comment (jrfnl) [inline README.md]: The spaces instead of tabs is something I did know ;-) Correct me if I'm wrong, but the example code uses spaces IIRC

File: README.md [hunk 4]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
 
 Please see “[Setting up PHP CodeSniffer in Visual Studio Code](https://tommcfarlin.com/php-codesniffer-in-visual-studio-code/)”, a tutorial by Tom McFarlin.
 
+
+## Running your code through WPCS automatically using CI tools
+
+### [Travis CI](https://travis-ci.org/)
+
+To integrate PHPCS with WPCS with Travis CI, you'll need to install both `before_install` and add the run command to the `script`.
+If your project uses Composer, the typical instructions might be different.
```
Comment (GaryJones) [inline README.md]: Different, how?

File: README.md (hunk 1 above)
Comment (ntwb) [inline README.md]: Yes, the example code does use spaces 👍

File: README.md [hunk 5]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
...
+
+If you use a matrix setup in Travis to test your code against different PHP and/or WordPress versions, you don't need to run PHPCS on each variant of the matrix as the results will be same.
+You can set an environment variable in the Travis matrix to only run the sniffs against one setup in the matrix.
+
+#### Travis CI example
+```bash
```
Comment (ntwb) [inline README.md]: yaml, not bash please

File: README.md [hunk 6]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
...
+language: php
+
+matrix:
+  include:
+    # Arbitrary PHP version to run the sniffs against.
+    - php: '7.0'
+      env: SNIFF=1
+
```
Comment (ntwb) [inline README.md]: There's ~10 spaces on this line, please remove them

File: README.md [hunk 7]
```diff
@@ -133,6 +133,49 @@ sublime-phpcs is insanely powerful, but if you'd prefer automatic linting, [Subl
...
+  - if [[ "$SNIFF" == "1" ]]; then git clone -b master --depth 1 https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards.git $SNIFFS_DIR; fi
+  # Set install path for WordPress Coding Standards.
+  - if [[ "$SNIFF" == "1" ]]; then $PHPCS_DIR/scripts/phpcs --config-set installed_paths $SNIFFS_DIR; fi
+  # After CodeSniffer install you should refresh your path.
+  - if [[ "$SNIFF" == "1" ]]; then phpenv rehash; fi
+
+script:
+  # Run against WordPress Coding Standards.
+  # If you use a custom ruleset, change `--standard=WordPress` to point to your ruleset file,
+  # for example: `--standard=wpcs.xml`.
```
Comment (GaryJones) [inline README.md]: Custom rulesets would usually be defined in `phpcs.xml` or the older `ruleset.xml`. Could we specify one of those instead of `wpcs.xml` (which, to me, thinks it might hold some special powers being named that way).

File: README.md (hunk 4 above)
Comment (jrfnl) [inline README.md]: > I don't use Composer much myself so didn't feel qualified to write it up, especially as I (probably mistakenly) thought it depended on if and if so, how phpcs and wpcs are defined in the composer.json file of a project.

https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/712#issuecomment-269378913

File: README.md (hunk 7 above)
Comment (jrfnl) [inline README.md]: I actually named it `wpcs.xml` on purpose as a `phpcs.xml` file will be picked up automatically if it's in the root of a project and doesn't need to be provided as an argument and IMHO explaining how that all works is outside the scope of this snippet of sample code.

File: README.md (hunk 6 above)
Comment (jrfnl) [inline README.md]: Fixed

File: README.md (hunk 5 above)
Comment (jrfnl) [inline README.md]: Fixed
